ts/index.ts

```diff
@@ -460,14 +460,28 @@ class Gossipsub extends Pubsub {
    * @returns {Promise<void>}
    */
   async _processRpcMessage (msg: InMessage): Promise<void> {
+    const canonicalMsgIdStr = await this.isSeen(msg)
+    // if not seen
+    if (canonicalMsgIdStr) {
+      // put in cache
+      this.seenCache.put(canonicalMsgIdStr)
+
+      await this.score.validateMessage(canonicalMsgIdStr)
+      await super._processRpcMessage(msg)
+    }
+  }
+
+  /** Return undefined if we've seen the message, fastMsgIdStr if not seen */
+  async isSeen (msg: InMessage): Promise<string | undefined> {
```

Comment:
The name is very confusing since responding to the question "is seen?" you expect truthy to mean yes, falsy to mean no.  What about a longer but more correct name like `getMsgIdIfNotSeen`.